### ISSUE TITLE
[Expo SDK 44][Jest] Cannot test components with jest-expo/ios or jest-expo/android

### ISSUE THREAD

**Issue by WNemencha** [expo/expo#18008]
### Summary

Hello,

I was trying to setup support for jest testing of components/providers in a React Native and it keeps throwing this error message at me (which I google'd/ddg'ed already, no results) so I'm coming to you for a potential fix/workaround.

Here's the error message:

```
FAIL   CLIENT:IOS  src/payments/provider.spec.tsx
  ● Test suite failed to run

    AssertionError [ERR_ASSERTION]: Unexpected: Config `_internal.projectRoot` isn't defined by expo-cli, this is a bug.

      at assertInternalProjectRoot (node_modules/jest-expo/node_modules/@expo/config-plugins/src/utils/plugin-resolver.ts:93:3)
          at Array.reduce (<anonymous>)
      at readExpoConfig (node_modules/jest-expo/src/preset/createMockConstants.js:50:10)
      at createMockConstants (node_modules/jest-expo/src/preset/createMockConstants.js:19:22)

 FAIL   CLIENT:ANDROID  src/payments/provider.spec.tsx
  ● Test suite failed to run

    AssertionError [ERR_ASSERTION]: Unexpected: Config `_internal.projectRoot` isn't defined by expo-cli, this is a bug.

      at assertInternalProjectRoot (node_modules/jest-expo/node_modules/@expo/config-plugins/src/utils/plugin-resolver.ts:93:3)
          at Array.reduce (<anonymous>)
      at readExpoConfig (node_modules/jest-expo/src/preset/createMockConstants.js:50:10)
      at createMockConstants (node_modules/jest-expo/src/preset/createMockConstants.js:19:22)
```

I'm also doing unit tests of reducers/selectors using the `jest-expo/node` preset (separate Jest projects for testing the state) and those tests pass just fine, so i don't understand what's going on here. See my `jest.config.ts` and `package.json` files bellow for version informations:

```ts
// jest.config.ts
const NODE_MODULES_TO_TRANSFORM = [
  "react-native",
  "jest-react-native",
  "@react-native",
  "@react-native-community",
  "expo-constants",
  "expo-modules-core",
  "expo-secure-store",
  "expo-random",
  "expo(nent)?",
  "@expo(nent)?/.*",
  "@expo-google-fonts/.*",
  "react-navigation",
  "@react-navigation/.*",
  "@unimodules/.*",
  "unimodules",
  "react-native-svg",
];

const projectBase = {
  clearMocks: true,
  collectCoverage: true,
  coverageDirectory: "coverage",
  coverageProvider: "v8",
  coverageReporters: ["json-summary", "text", "lcov"],
  coverageThreshold: {
    global: {
      statements: 90,
      branches: 75,
      functions: 85,
      lines: 90,
    },
  },
  globals: {
    __DEV__: false,
  },
  moduleFileExtensions: ["ts", "tsx", "js", "json", "jsx", "node"],
  moduleNameMapper: {
    "^test/utils$": "<rootDir>/src/utils/test/index",
    "@react-native-firebase/app":
      "<rootDir>/__mocks__/@react-native-firebase/app.ts",
    "@react-native-firebase/auth":
      "<rootDir>/__mocks__/@react-native-firebase/auth.ts",
    "@react-native-firebase/firestore":
      "<rootDir>/__mocks__/@react-native-firebase/firestore.ts",
    "@react-native-firebase/messaging":
      "<rootDir>/__mocks__/@react-native-firebase/messaging.ts",
  },
  testEnvironmentOptions: {},
  transformIgnorePatterns: [
    `/node_modules/(?!(${NODE_MODULES_TO_TRANSFORM.join("|")}))`,
    "\\.pnp\\.[^\\/]+$",
  ],
};

const mobileProjectBase = {
  // setupFilesAfterEnv: ["@testing-library/jest-native/extend-expect"],
  testMatch: ["!src/state/*", "**/?(*.)+(spec).[jt]s?(x)"],
};

export default {
  projects: [
    {
      ...projectBase,
      displayName: "STATE",
      preset: "jest-expo/node",
      testMatch: ["<rootDir>/src/state/**/?(*.)+(test).[jt]s"],
    },
    {
      ...projectBase,
      ...mobileProjectBase,
      displayName: "CLIENT:ANDROID",
      preset: "jest-expo/android",
    },
    {
      ...projectBase,
      ...mobileProjectBase,
      displayName: "CLIENT:IOS",
      preset: "jest-expo/ios",
    },
  ],
};
```

```json
// package.json
{
  "dependencies": {
    "expo": "^44.0.6",
    "react": "17.0.1",
    "react-native": "0.64.3",
  },
  "devDependencies": {
    "@babel/core": "~7.9.0",
    "@babel/preset-typescript": "^7.16.7",
    "@jest/types": "^27.5.1",
    "@testing-library/jest-native": "^4.0.4",
    "@testing-library/react-native": "^9.2.0",
    "@types/jest": "^28.1.3",
    "@types/node": "^17.0.25",
    "@types/react": "~17.0.21",
    "@types/react-native": "^0.64.3",
    "babel-jest": "^27.5.1",
    "babel-preset-expo": "9.0.2",
    "expo-cli": "~5.4.12",
    "jest": "^27.5.1",
    "jest-environment-node": "^27.5.1",
    "jest-expo": "^44.0.1",
    "react-test-renderer": "^17.0.1",
    "ts-jest": "27.1.5",
    "typescript": "^4.5.4"
  }
}
```

### What platform(s) does this occur on?

Android, iOS

### SDK Version

44

### Environment

    expo-env-info 1.0.4 environment info:
    System:
      OS: macOS 12.3
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 16.14.0 - ~/.nvm/versions/node/v16.14.0/bin/node
      Yarn: 1.9.2 - ~/.yvm/shim/yarn
      npm: 8.3.1 - ~/.nvm/versions/node/v16.14.0/bin/npm
      Watchman: 2022.05.16.00 - /usr/local/bin/watchman
    Managers:
      CocoaPods: 1.11.2 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 21.4, iOS 15.4, macOS 12.3, tvOS 15.4, watchOS 8.5
    IDEs:
      Android Studio: 4.1 AI-201.8743.12.41.7042882
      Xcode: 13.3/13E113 - /usr/bin/xcodebuild
    npmPackages:
      babel-preset-expo: 9.0.2 => 9.0.2 
      expo: ^44.0.6 => 44.0.6 
      react: 17.0.1 => 17.0.1 
      react-dom: 17.0.1 => 17.0.1 
      react-native: 0.64.3 => 0.64.3 
      react-native-web: 0.17.1 => 0.17.1 
      react-navigation: ^4.4.4 => 4.4.4 
    npmGlobalPackages:
      expo-cli: 5.3.0
    Expo Workflow: bare

### Reproducible demo

The `payments.provider.ts` file looks like this (nothing crazy yet):

```ts
import React from "react";
import { Text } from "react-native";
import { render } from "@testing-library/react-native";

import { PaymentsProvider } from "./provider";

describe("PaymentsProvider", () => {
  it(`should render correctly`, async () => {
    // Given
    const TestComponent = () => {
      const isPurchaserSubscribed = false;
      return (
        <Text testID={"isPurchaserSubscribed"}>
          Plan: {isPurchaserSubscribed ? "Pro" : "Free"}
        </Text>
      );
    };

    // When
    const { findByTestId } = render(
      <PaymentsProvider>
        <TestComponent />
      </PaymentsProvider>
    );

    const subscriptionStatus = await findByTestId("subscriptionStatus");

    // Then
    expect(subscriptionStatus).toMatchSnapshot();
  });
});

```

**Comment by WNemencha** [comment]
Update, figured out I was using `expo-cli@5.3.0` through global packages (in nvm folder), removing those global modules and installing `expo-cli@5.4.12` in my devDependencies changed the stacktrace a bit, but still the same error.

```
 FAIL   CLIENT:IOS  src/payments/provider.spec.tsx
  ● Test suite failed to run

    AssertionError [ERR_ASSERTION]: Unexpected: Config `_internal.projectRoot` isn't defined by expo-cli, this is a bug.

      at assertInternalProjectRoot (node_modules/@expo/config-plugins/src/utils/plugin-resolver.ts:130:44)
      at withStaticPlugin (node_modules/@expo/config-plugins/build/plugins/withStaticPlugin.js:87:53)
      at node_modules/@expo/config-plugins/src/plugins/withPlugins.ts:18:5
          at Array.reduce (<anonymous>)
      at withPlugins (node_modules/@expo/config-plugins/src/plugins/withPlugins.ts:12:81)
      at withConfigPlugins (node_modules/@expo/config/build/plugins/withConfigPlugins.js:36:47)
      at fillAndReturnConfig (node_modules/@expo/config/src/Config.ts:105:39)
      at getConfig (node_modules/@expo/config/src/Config.ts:165:5)
      at readExpoConfig (node_modules/jest-expo/src/preset/createMockConstants.js:50:10)
      at createMockConstants (node_modules/jest-expo/src/preset/createMockConstants.js:19:22)
```

**Comment by WNemencha** [comment]
@brentvatne Hey, thanks for the reply! I did install expo-cli after starting to see the issue in the process of "tying to get package.json versions to align (jest, babel-jest, jest-expo, @types/jest, etc)" but adding expo-cli to the local node_modules did not change the output.

To confirm that I just removed it from the package.json file, uninstalled my node version completely (through nvm) then reinstalled it, and then executed this command: `npm install -g "expo-cli@5.4.12"` which correctly added the cli in `/Users/wnemencha/.nvm/versions/node/v16.14.0`. I ran `expo-cli --version` which output `5.4.12` so it seems fine.

Yet when I run the tests again after all this; same error message :/
What is super-weird too is that my other Jest' project uses the same preset (`jest-expo/node`) only a different "import" from it so I'd expect it to also be in a failing state if it was linked to 'expo-cli' somehow. See the screenshot:

<img width="874" alt="image" src="https://user-images.githubusercontent.com/40295432/176252449-43f8257c-dd1a-482c-8983-d8d4e29cbb83.png">

I've investigated through Expo's codebase a bit and found this which seems related/a way to mock it, do you think its safe to take that mock? Asking that because its labeled as `_internal` so the naming couldn't be more explicit :/

-> https://github.com/expo/expo-cli/blob/7229a3153a2ad9f3b2b35569929a1aa0110e2699/packages/config-plugins/src/plugins/__tests__/static-plugins-test.ts#L7-L18

**Comment by brentvatne** [comment]
@WNemencha - thanks for the follow up. can you share a minimal reproducible example? https://stackoverflow.com/help/minimal-reproducible-example

**Comment by WNemencha** [comment]
@brentvatne I will try to setup one this week-end, in the meantime I've switched to using `jest/react-native` preset in combination with importing the setup+mocks files from `jest-expo` into project's own `__mocks__` folder + setup file ;)

**Comment by WNemencha** [comment]
Update: i've not been able to make the reproduction this weekend, will try this week

**Comment by UlasRenasOrdu** [comment]
Check your app.json file!!!

![Image](https://github.com/user-attachments/assets/ca51c039-02e1-4bde-980b-473985b23bb7)

I had the same problem yesterday and it took 4 hours of my time.Luckily I found the problem.
I added the library "react-native-reanimated": "^3.16.2" and added it to plugins to use it and then this error started to occur.
So check the Plugins thoroughly. Especially if there is something you added recently, remove it and try running it.

I understand that if there is an unnecessary definition, it gives this error. I hope you solve the problem as soon as possible.
